optuna/distributions.py

```diff
@@ -450,6 +453,12 @@ def _asdict(self) -> Dict:
         return d
 
 
+def _nanequal(value1: Any, value2: Any) -> bool:
```

Comment:
[nits] Not a strong opinion but how about naming this `_param_value_equal()`? `_nanequal` sounds like just checking both of two values are nan or not for me.